src/DynamoCore/DSEngine/AstBuilder.cs

```diff
@@ -83,6 +93,80 @@ public static IEnumerable<NodeModel> TopologicalSort(IEnumerable<NodeModel> node
             return sortedNodes.Where(nodeModels.Contains);
         }
 
+        /// <summary>
+        /// BFS and post-order traverse to ensure all inputs are visited in
+        /// order and before downstream node.
+        /// </summary>
+        /// <param name="node"></param>
+        /// <param name="nodeFlags"></param>
+        /// <param name="sortedNodes"></param>
+        private static void BFSTraverse(
+            NodeModel node, 
+            Dictionary<NodeModel, MarkFlag> nodeFlags, 
+            Queue<NodeModel> sortedNodes)
+        {
```

Comment:
It seems like a `null node` is being inherently handled, but is that why we do not have any `null` checks against `node`? I'm just looking at `t.Item2` and feel afraid that it might be `null`.